R/hp_datatree.R

```diff
@@ -0,0 +1,141 @@
+dt_root <- function(x) {
+  names(x$Get("level", filterFun = data.tree::isRoot))
+}
+
+dt_root_df <- function(x) {
+  x <- dt_root(x)
+  data.frame(
+    from = x,
+    to = x,
+    stringsAsFactors = FALSE
+  )
+}
+
+dt_collect <- function(x) {
```

Comment:
Say if there is a dataset with a lot of columns, would it be worth it to only collect the fields specified in `attributes` param from `hp_datatree` (i.e. add attributes as a param here and lapply through attributes as opposed to fieldsAll)?

Reply:
Hm, this is an interesting idea. I will play around with it.

Reply:
On second thought, how do you feel about me filing an enhancement issue for this? If we run into an example that is terribly slow because of unneeded columns for example.

Reply:
Fine by me!